pypowerwall/__init__.py

```diff
@@ -606,3 +681,52 @@ def get_time_remaining(self) -> Optional[float]:
             The time remaining in hours
         """
         return self.client.get_time_remaining()
+
+    def _validate_init_configuration(self):
+
+        # Basic user input validation for starters. Can be expanded to limit other parameters such as cache
+        # expiration range, timeout range, etc
+
+        # Check for valid hostname/IP address
+        if (self.host and (
+                not isinstance(self.host, str) or
+                (not IPV4_6_REGEX.match(self.host) and not HOST_REGEX.match(self.host))
+        )):
+            raise PyPowerwallInvalidConfigurationParameter(f"Invalid powerwall host: '{self.host}'. Must be in the "
+                                                           f"form of IP address or a valid form of a hostname or FQDN.")
+
+        # If cloud mode requested, check appropriate parameters
+        if self.cloudmode:
+            # Ensure email is set and syntactically correct
+            if not self.email or not isinstance(self.email, str) or not EMAIL_REGEX.match(self.email):
+                raise PyPowerwallInvalidConfigurationParameter(f"A valid email address is required to run in "
+                                                               f"cloud mode: '{self.email}' did not pass validation.")
+
+            # Ensure we can write to the provided authpath
+            dirname = self.authpath
+            if not dirname:
+                dirname = '.'
+            self._check_if_dir_is_writable(dirname)
+        # If local mode, check appropriate parameters, too
+        else:
+            # Ensure we can create a cachefile
+            dirname = os.path.dirname(self.cachefile)
+            if not dirname:
+                dirname = '.'
+            self._check_if_dir_is_writable(dirname)
+
+    @staticmethod
+    def _check_if_dir_is_writable(dirpath):
```

Comment:
This needs to be tested on Windows, if it causes issues we can fix it or remove the whole thing or the particular piece of it which causes issues.